### PR TITLE
Vendor asset fix

### PR DESCRIPTION
**Pull request by mullenpaul**
## About The Pull Request

The vendor TGUI migration had a few hickups and has had mixed feedback, this PR addresses this.

The major effort here is resolving the icon issue in the ui. The icons are generated during initialization. During testing, the icons were generated as there was a client connected to the server. When a client remotely connects, after server initialisation, the icons aren't generated on the server. During initialisation, the assets are now directly created and registered with the asset system.

Furthermore, the assets were not listed in the 'preload' var in the asset transport mechanism. So when a remote player connects, the icon assets generated during initialisation are not streamed over.

Both of these issues are resolved.

There is also some UI tweaking, taking some of the constructive criticism I have received, see changelog for details.

image

## Why It's Good For The Game

Bugfixes and better UX

## Changelog

:cl:
add: added mechanism to dynamically add assets to the preload asset list
fix: vendor ui icons now rendering for remote clients
qol: Improved vendor ui layout to reduce deadspace
qol: vend icons now styled with recommended / mandatory colouring
qol: Description box for vendor UI now only shows when hovering over info icon
/:cl:

### CODE REVIEW

File: code/modules/asset_cache/asset_list_items.dm [hunk 1]
```diff
@@ -127,6 +127,32 @@
 			assets[filename] = fcopy_rsc(path + filename)
 	..()
 
+/datum/asset/simple/dynamic_icons
```
Comment (mullenpaul) [inline code/modules/asset_cache/asset_list_items.dm:130]: I'll add in some documentation here for future generations

**Comment by JohnDoe50974** [comment]
I'd highly prefer the old button style was kept, but the button placement is indeed much better. +1!

**Comment by mullenpaul** [comment]
Marked as draft, need to test this with the antag snowflake vendors

**Review by stanalbatross — APPROVED** [review]

**Comment by stanalbatross** [comment]
should fix this issue

**Pull request merged**